Add Zadarma SMS provider and event handler

KitaroAI opened the pull request:
- Added `provider-Zadarma.php` to support Zadarma SMS
- Updated `public/provider.php` to handle SMS webhook routing
- Tested with Zadarma DID and Groundwire for inbound SMS

KitaroAI committed:
Create provider-Zadarma.php

for those that want sms with Zadarma

KitaroAI committed:
Update provider.php

Update provider.php to handle Zadarma SMS event

KitaroAI commented:
Thanks for reviewing. Let me know if you’d like me to document this, add test cases, or adjust anything to match project standards. Happy to help however I can.

billsimon commented on providers/provider-Zadarma.php:
```diff
@@ -0,0 +1,124 @@
+        if ($_SERVER['REQUEST_METHOD'] === "POST") {
+            $postdata = $_POST;
+            // Log incoming post for debug (remove/comment in production)
+            file_put_contents('/tmp/sms_webhook_debug.log', date('c') . " " . print_r($postdata, true) . "\n", FILE_APPEND);
```
Please remove debug code for release

billsimon commented on providers/provider-Zadarma.php:
```diff
@@ -0,0 +1,124 @@
+                $text = isset($result['text']) ? $result['text'] : '';
+
+                // Save to debug as well
+                file_put_contents('/tmp/sms_webhook_debug.log', "SMS From: $from To: $to Msg: $text\n", FILE_APPEND);
```
Please remove this debug code also for release

billsimon commented on public/provider.php:
```diff
@@ -1,4 +1,6 @@
 <?php
+if (isset($_GET['zd_echo'])) exit($_GET['zd_echo']); //verification request from Zadarma with the value
```
I believe this Zadarma-specific code should be elsewhere since this provider.php is used for all providers. The webhook should be hitting the provider-Zadarma.php file so this logic should be included there instead.

KitaroAI committed:
Cleanup Zadarma provider per maintainer review

- Removed debug logging and file writes
- Reverted public/provider.php changes
- Added zd_echo verification inside provider
- Inbound SMS handled via receiveMessage only
- No core file modifications

KitaroAI committed:
Restore public/provider.php to upstream version
